src/main/java/com/selfxdsd/todos/PuzzlesApi.java

```diff
@@ -89,27 +88,25 @@ public PuzzlesApi(
     )
     public ResponseEntity<String> reviewPuzzles(
         @PathVariable final String provider,
-        @PathVariable final String owner, @PathVariable final String name,
+        @PathVariable final String owner,
+        @PathVariable final String name,
         @RequestBody final String payload
-    ) {
+    ) throws Exception {
+        final Self selfCore = this.beanFactory.getBean(Self.class);
```

Comment:
since SelfCoreComponent is prototype, each time we request for a Self bean, a new one will be created.